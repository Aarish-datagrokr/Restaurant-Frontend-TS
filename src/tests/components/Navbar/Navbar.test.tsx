import React, { Component } from 'react';
import {shallow} from 'enzyme';
import Navbar from '../../../components/Navbar/Navbar';

const setState = jest.fn();
const useStateSpy:any = jest.spyOn(React, "useState");
useStateSpy.mockImplementation((initialState:Boolean) => [initialState, setState]);
const wrapper = shallow<Component>(<Navbar />);

test('Should test Navbar component', () => {
    expect(wrapper).toMatchSnapshot();
});

it("should toggle isNavbarExpanded on click", () => {
    const newInputValue:Boolean = true;
    wrapper
      .find('#hamburger')
      .simulate("click", { target: { value: newInputValue } });
  });

  it("should toggle isNavbarExpanded on click", () => {
    const newInputValue = true;
    wrapper
      .find('#add')
      .simulate("click", { target: { value: newInputValue } });
  });

  it("should toggle isNavbarExpanded on click", () => {
    const newInputValue = true;
    wrapper
      .find('#update')
      .simulate("click", { target: { value: newInputValue } });
  });

  it("should toggle isNavbarExpanded on click", () => {
    const newInputValue = true;
    wrapper
      .find('#delete')
      .simulate("click", { target: { value: newInputValue } });
  });