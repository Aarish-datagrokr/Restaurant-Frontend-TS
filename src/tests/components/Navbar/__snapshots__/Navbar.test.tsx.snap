// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test Navbar component 1`] = `
<nav
  className="navigation"
>
  <Link
    className="brand-logo"
    to="/"
  >
    <img
      src="https://www.pinclipart.com/picdir/middle/86-862588_book-fair-clip-art.png"
      style={
        Object {
          "height": 50,
          "width": 60,
        }
      }
    />
  </Link>
  <button
    className="hamburger"
    id="hamburger"
    onClick={[Function]}
  >
    <svg
      className="h-5 w-5"
      fill="white"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM9 15a1 1 0 011-1h6a1 1 0 110 2h-6a1 1 0 01-1-1z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <div
    className="navigation-menu"
  >
    <ul>
      <li>
        <Link
          id="add"
          onClick={[Function]}
          to="/add"
        >
          Add Booking
        </Link>
      </li>
      <br />
      <li>
        <Link
          id="update"
          onClick={[Function]}
          to="/update"
        >
          Update Booking
        </Link>
      </li>
      <br />
      <li>
        <Link
          id="delete"
          onClick={[Function]}
          to="/delete"
        >
          Delete Booking
        </Link>
      </li>
      <br />
    </ul>
  </div>
</nav>
`;
